### DISCUSSION TITLE
Xdrip show value above 400 (25)

### DISCUSSION THREAD

**Discussion by Memozone** [NightscoutFoundation/xDrip#2945] [Q&A]
How i can see the value above 400(25) to know the trend if it start to go down or still..in my phone it show high then flat line
![inbound7130844636011903671](https://github.com/NightscoutFoundation/xDrip/assets/139771853/e8e12a14-03a9-44d8-8e48-703e18473e09)

**Comment by Navid200** [comment]
There is no way.  I have removed that portion because I thought we were wasting precious limited screen space for no good reason.
But, I could have been mistaken, and thanks for bringing it up.

Would you please help me see why that matters?
How would you act differently if you know that your blood glucose is not 400 but 450?

Let me also ask you something else.  Where would you draw the line?  I am not saying we are going to change this.  But, let's say we change this from 400 to 410 and then, someone says why it it flat at 410?  So, my question is at what level will you say OK it makes no difference anymore if I am higher than that, I need to lower it and it makes no difference exactly what it is right now.

**Reply by Memozone** [reply]
For me my son on insulin pump sometime he take alote if sweet when i just insert the pump with non calculated card... i give him correction and want to see if its from the pump or just the sweet.. if the trend go down above the 400 and its go down so its work.. i can wait.. but if its above 400 and the trend is still go up i have to change it.. and belive me this happen alot with children DM1

**Reply by Memozone** [reply]
![Uploading Screenshot_20230717_144346_Firefox.jpg…]()
It well be like this

**Comment by Navid200** [comment]
There is no excuse for having any shaming here.  That is not my intent.  

If someone sees a very low reading, they act to raise their glucose.  Because the common goal is to be in range.  Likewise, if someone is high, they will act to lower glucose, again because they want to be in range.

It is very important to take the trend into account because it gives us a way of forecasting the future.  If the glucose is changing at a higher pace, we need to take that into account.

But, if you are over 400, after you come below it, you will be able to see the trend.
I cannot think of a scenario where not being able to see the trend above 400 would cause a dangerous consequence.

Taking too much insulin is dangerous.  No one should take too much insulin.
Diabetes treatment requires consultation with a medical team.   Please don't consider anything I say as medical advice.

I have tried to explain why I have considered readings above 400 irrelevant.  If that is a mistake, I would like to understand why.

**Reply by Memozone** [reply]
So i want to only see if the trend go down or up to be sure.. not for treatment or take another correction i use it with blood test ..but its useful for me to see the trend above 400

**Comment by jonprogers** [comment]
I'm not sure which data source you are using although I currently use G6 and believe 400 is the highest reading that can be provided by a Dexcom CGM so the limit could be a result of the hardware data source and not a decision made by xDrip...

I also echo the comments from Navid200 - my treatment decision would be pretty much the same if I'm at 400 or 500.

**Comment by PhilTU** [comment]
Does the trend arrow next to the blood glucose value not inform you about the direction of travel ?

**Reply by Memozone** [reply]
Above 400 arrow only flat no up and down like the picture i post

**Comment by Navid200** [comment]
We did not draw a line at 400.  Dexcom did.  
Other sensors do produce readings greater than 400.
We used to show them all.  At some point, we decided to limit how much we change the vertical (y) axis scale in order to show large values and limited values greater than 400 to 400.

My concern with showing the trend arrow for such values is the accuracy of readings greater than 400.  What is positive about that is that it will not change the axis scale.

So, we may need to talk about this and decide if something can be done.

**Comment by Memozone** [comment]
For me my son on insulin pump sometime he take alote of sweet when i just insert the pump with non calculated carb... i give him correction and want to see if its from the pump or just the sweet.. if the trend go down above the 400 and its go down so its work.. i can wait.. but if its above 400 and the trend is still go up i have to change the pump... and belive me this happen alot with children with DM1..
So i want to only see if the trend go down or up to be sure.. not for treatment or take another correction i do it with blood test ..but its useful for me to see the trend above 400..and it will be like this photo (i download it from Internet not mine)
![Screenshot_20230717_144346_Firefox](https://github.com/NightscoutFoundation/xDrip/assets/139771853/25d08071-d870-4820-86c3-bc0c982e5051)

**Comment by psonnera** [comment]
@tzachi-dar as Navid says, Dex locks anything above 400 and so does Abtt. I'm wondering if it's not something in the OOP that allows results above 400 like the Blukon picture above. As posted on FB I've seen values of 415 being transmitted to AAPS. What do you think?
On the other hand, looking a Nightscout, the same 400 limit applies, so it's consistent. Not suggesting anything, just curious.

**Comment by Navid200** [comment]
We made the change on February 11, 2023.
https://github.com/NightscoutFoundation/xDrip/pull/2652

This change makes no difference for Dexcom G6 or G7.
If you use a device that shows values greater than 400 mg/dL, for values greater than 400, you now see no trend arrow or you see a flat trend arrow.
We are not going to undo that change we made to the axis.  We made that change because there were complaints.

I can think of two possibilities in order to maintain the trend arrow.

1- We can show 400 but also show a trend arrow representing the trend arrow if we had not made the change.
2- We can keep the axis range limited to 400.  But, show the current value before being capped at 400 and also show the trend arrow that you would have seen if we had not made the change.

Neither of those two options is acceptable in my opinion because they will be very confusing considering the change in the axis range.

The best choice in my opinion is to accept that xDrip provides no values greater than 400.  If seeing values greater than 400 is a requirement for you, you can use the official app or device for your sensor.

**Reply by Memozone** [reply]
I hope there is an option to turn on and off so it will be helpful for some people and the others can turn it off so they will not cofused... but anyway thanks for your help and thanks for your try and reply...appreciate

**Reply by Navid200** [reply]
What sensor do you use?  Are you sure it provides values greater than 400?

Regardless, the FDA disagrees with you.  The FDA has approved Dexcom G6, which does not provide any readings over 400, for looping.  If the FDA thought that it would be confusing, they would not have approved it.

**Reply by Memozone** [reply]
I use libre 2

**Comment by Miustone** [comment]
Sorry to bring this up again but I would really love to get at least an Option for the y Axis to show values and lines up to 500. My Libre 3 does show me Values up to 500 and it makes a huge difference for me if I take 1ie or 2ie of My FIASP more. Besides that are most Screens today not 16:9 anymore and there is more than enough space on My Poco F6 and Galaxy S22 Ultra to show everything...

On top, why does the scale go down to 0 from 450mg/dl (under 50mg/dl)? Most likely I don't need to know if I am under 50 at all but if I'll go down there, based on to much correction with FIASP when over 400...

So Please. Let us set the scale ourselves on the y Axis! (Longtime G6 and Xdrip+ User with Novopens)

**Reply by Navid200** [reply]
You can customize the y axis range:  https://navid200.github.io/xDrip/docs/Display/GlucoseRange.html  

However, 400 is the highest you can set it to.  
Technically, I can add a 500 option.  
But, I have two concerns with that.

If a device reports a value greater than 400, it is not going to be very accurate.  I don't think you will be flat at such a high value and you won't be calibrating at such a high value.

But, more importantly, my main concern is what the user is going to do.  
Someone may be under the impression that if they have a very high value, they must inject a lot of insulin.  
I am not a doctor and am not going to tell you what to do.  
But, my concern is that someone may inject too much insulin not realizing that the more insulin they inject, the faster the glucose is going to drop.  And what you see on the screen of a CGM is not really your blood glucose;  it is your interstitial fluid glucose multiplied by a calibration factor.  But, the timing corresponds to your interstitial fluid, which is delayed by about 10 minutes.    So, by the time the CGM shows low, your blood glucose may be much lower, specially if the slope is too steep.

My worry is that someone is going to take too much insulin and have a hard time keeping their glucose from overcorrecting to drop too low.

**Comment by PhilTU** [comment]
Libre 3 device caps the y axis...

FS-Libre-US: Understanding Sensor Readings (Reader) Note: While Sensor glucose readings are gathered in the System range of 40-500 mg/dL, the graph display range is 0-350 mg/dL for ease of review on screen.

**Comment by marciogranzotto** [comment]
I don't think we should limit the users. At least we should provide an advanced option to change the max-y value.

**Comment by Miustone** [comment]
I think following the Sensors Limits natively would make the most Sense. There is a Reason for using Xdrip+ instead of the OG Apps and one should be a Scale that can reflect the actual Readings. My Alarms already show values like 445mg/dl (Xdrip+ with Libre 3 Webfollower) so why shouldn't it do the same in the App, Lockscreen and Widget too? Libre 3 capping at 350 but displaying up to 500 is just bad Design in My Opinion.  The upper part of the App is just a waste of Space...

**Comment by Miustone** [comment]
Example of wasted Space
![Screenshot_20240731_114632](https://github.com/user-attachments/assets/ca31e22b-d726-42ce-a4d0-5bd437444056)

**Reply by psonnera** [reply]
I'm afraid the issue here is not the graph maximum scale. But please don't get offended by this comment. I tend to agree with Navid, knowing ADA range recommendations. Above Y axis limit you don't need a CGM and should use a blood sample to correct back to target hence avoid taking decisions using inaccurate measurement.

**Reply by PhilTU** [reply]
Another example of wasted space, but design decisions about Librelink's lack of y-axis scaling are probably outside our scope 
![wasted](https://github.com/user-attachments/assets/19d611fe-8392-4e1c-9b1b-fd4ea297f501)

**Comment by Miustone** [comment]
I am always counter measuring for Calibration with Accu Check Mobile which is always on point. Usually my Readings are like 10 minutes behind up and down. Dexcom also literally says to dont make decisions on what the Sensors say but  based on Your feel and actual Bloodglucose level... So i don't think that's a point at all.
My Point is for better analysis and a wider range for the predictions with Insulin working... 
![Screenshot_20240731_120631_xDrip+](https://github.com/user-attachments/assets/70f7c0f3-0d8b-42ac-9301-147cf23a0008)

**Reply by Navid200** [reply]
I'm going to take an action here:

Find out if the predictive simulations use real values versus the clamped max values.

My expectation is that predictive simulations don't care about what is shown on screen.  They use readings even if they are very high.  But, I need to confirm that.

**Reply by Navid200** [reply]
It is possible that a doctor may tell someone to keep their blood glucose at high values.
We cannot give medical advice here.  We can talk about the app only.  
With respect to the app, since the majority keep their blood glucose at significantly lower values than 400, we have made xDrip suitable for the majority to minimize variations in the slopes of the curves shown on screen.  
  
---  

**xDrip is not a medical app**  
**You need a doctor with or without xDrip**  
  
Everyone needs medical advice even if they don't have diabetes.  But, specially if they do have diabetes.
No one should use xDrip and think that they don't need a doctor as a result.  That couldn't be farther from the truth.  

Insulin can kill you if you don't use it correctly.  Even if it doesn't kill you, it can seriously hurt you.  
High blood glucose can cause serious long-term health complications.  
   
If you have diabetes, please see a doctor and follow their advice to safely control your blood glucose.
